http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.java

```diff
@@ -140,12 +141,23 @@ private RequestCallable(HttpURLConnection connection, HttpExecuteRequest request
         public HttpExecuteResponse call() throws IOException {
             connection.connect();
 
-            request.contentStreamProvider().ifPresent(provider ->
-                    invokeSafely(() -> IoUtils.copy(provider.newStream(), connection.getOutputStream())));
+            // Only allow sending a body if one of the below methods. Matches ApacheHttpRequestFactory#createApacheRequest
+            SdkHttpMethod sdkHttpMethod = request.httpRequest().method();
+            if (sdkHttpMethod == SdkHttpMethod.PATCH ||
+                sdkHttpMethod == SdkHttpMethod.POST ||
+                sdkHttpMethod == SdkHttpMethod.PUT) {
+
+                // TODO: Does this need to wrap like in ApacheHttpRequestFactory#wrapEntity?
+                request.contentStreamProvider().ifPresent(provider ->
```

Comment:
I wonder if the `contentStreamProvider` should actually return null if it's an unsupported `SdkHttpMethod` - otherwise every implementation of an HTTP client has to have this logic.

Reply:
It seems it does, the test harness always passed it, so fixed the test harness